src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -447,27 +448,35 @@ trait Infer extends Checkable {
      *  @return map from tparams to inferred arg, if inference was successful, tparams that map to None are considered left undetermined
      *    type parameters that are inferred as `scala.Nothing` and that are not covariant in `restpe` are taken to be undetermined
      */
-    def adjustTypeArgs(tparams: List[Symbol], tvars: List[TypeVar], targs: List[Type], restpe: Type = WildcardType): AdjustedTypeArgs.Result  = {
-      val buf = AdjustedTypeArgs.Result.newBuilder[Symbol, Option[Type]]
+    def adjustTypeArgs(tparams: List[Symbol], tvars: List[TypeVar], targs: List[Type], restpe: Type = WildcardType): AdjustedTypeArgs  = {
```

Comment:
I think that, much like your suggestion in [another PR](https://github.com/scala/scala/pull/7875#issuecomment-473155072), here `tparams` is equal to `tvars1 map (_.origin.typeSymbol)`, so perhaps we could omit that parameter. Note, however, that it would not save its allocation where it is called from.